src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let stateClone = Object.assign({}, state);
+
+  actions.forEach(function (element) {
+    if (element.type === 'clear') {
+      stateClone = {};
+      stateHistory.push({ ...stateClone });
+    }
+
+    if (element.type === 'addProperties') {
+      stateClone = { ...stateClone, ...element.extraData };
+      stateHistory.push({ ...stateClone });
```

Comment:
This comment applies to all cases where you push to stateHistory: you're repeating the same action, which violates the DRY principle. Move the push operation to the end of the loop cycle, outside of the switch block.